File: fabric/src/main/java/dev/obscuria/fragmentum/fabric/FabricPlatform.java
package dev.obscuria.fragmentum.fabric;

import dev.obscuria.fragmentum.Platform;
import net.fabricmc.api.EnvType;
import net.fabricmc.loader.api.FabricLoader;

public class FabricPlatform implements Platform {

    @Override
    public String getEnvironmentName() {
        return "Fabric";
    }

    @Override
    public boolean isModLoaded(String modId) {
        return FabricLoader.getInstance().isModLoaded(modId);
    }

    @Override
    public boolean isClient() {
        return FabricLoader.getInstance().getEnvironmentType() == EnvType.CLIENT;
    }

    @Override
    public boolean isDedicatedServer() {
        return FabricLoader.getInstance().getEnvironmentType() == EnvType.SERVER;
    }

    @Override
    public boolean isDevelopmentEnvironment() {
        return FabricLoader.getInstance().isDevelopmentEnvironment();
    }
}

File: fabric/src/main/kotlin/dev/obscuria/fragmentum/fabric/FabricPlatform.kt
package dev.obscuria.fragmentum.fabric

import dev.obscuria.fragmentum.api.Platform
import net.fabricmc.api.EnvType
import net.fabricmc.loader.api.FabricLoader

class FabricPlatform : Platform
{
    override fun getPlatformName(): String
    {
        return "Fabric"
    }

    override fun isModLoaded(modId: String): Boolean
    {
        return FabricLoader.getInstance().isModLoaded(modId)
    }

    override fun isClient(): Boolean
    {
        return FabricLoader.getInstance().environmentType == EnvType.CLIENT
    }

    override fun isDevelopmentEnvironment(): Boolean
    {
        return FabricLoader.getInstance().isDevelopmentEnvironment
    }
}